### ISSUE TITLE
ZMQ_EVENT_HANDSHAKE_FAILED not declared

### ISSUE THREAD

**Issue by lucaszanella** [zeromq/cppzmq#177]
I'm getting

```
/usr/local/include/zmq.hpp:866:18: error: 'ZMQ_EVENT_HANDSHAKE_FAILED' was not declared in this scope
             case ZMQ_EVENT_HANDSHAKE_FAILED:
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/zmq.hpp:869:18: error: 'ZMQ_EVENT_HANDSHAKE_SUCCEED' was not declared in this scope
             case ZMQ_EVENT_HANDSHAKE_SUCCEED:
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~

```

when I compile my project with `-DZMQ_BUILD_DRAFT_API=1` so  `-DZMQ_HAVE_POLLER=1`can turn on poller.

Was it meant to be like this?

**Comment by lucaszanella** [comment]
I can't find the definitions
```

  - ZMQ_EVENT_HANDSHAKE_SUCCEED
- ZMQ_EVENT_HANDSHAKE_FAILED
```

anywhere, just their mention in the `NEWS` file: https://github.com/zeromq/libzmq/search?utf8=%E2%9C%93&q=ZMQ_EVENT_HANDSHAKE_SUCCEED&type=

**Comment by sigiesec** [comment]
Which version of cppzmq (resp. the commit hash, since there are no version numbers) and libzmq are you using? The DRAFT API around the handshake events has changed between libzmq 4.2.2 and 4.2.3. Maybe you need to update zmq.hpp?

**Comment by lucaszanella** [comment]
My libzmq is 4.2.3 and cppzq 4.2.2 (since there is no 4.2.3). I'm using the release: https://github.com/zeromq/cppzmq/archive/v4.2.2.tar.gz for cppzmq, and https://github.com/zeromq/libzmq/releases/tag/v4.2.3 for libzmq

The zmq.hpp is defined to use the handshake macro if the version is after or equal 4.2.1: https://github.com/zeromq/cppzmq/blob/249b4a09a7f6446d30aabbc8158e54ed5754080f/zmq.hpp#L869

I found that it exists in 4.2.2:

https://github.com/zeromq/libzmq/blob/8d75d506b6f88265bc0fb57b93385b45e35cdd10/include/zmq.h#L566

I thought that the two latest versions for `libzmq` and `cppzmq` would work together, but it seems that `cppzmq` were not built for `4.2.3` yet.

I'm going to disable DRAFT then :(
(I just needed poller_t in zmq_cpp, but it seems is going to take time until its standardized)

**Comment by sigiesec** [comment]
As you say, there is no 4.2.3 release of cppzmq yet. You could use the version from master.
One of the owners would need to publish a release.
@bluca Can you release cppzmq 4.2.3?

**Comment by bluca** [comment]
done

**Comment by sigiesec** [comment]
I assume this problem is resolved by cppzmq 4.2.3, so I am closing this issue. Please reopen if problems persist.
